client/client.go

```diff
@@ -680,21 +679,9 @@ func (c *Client) init() error {
 		"reserved", reserved,
 	)
 
-	// Ensure cgroups are created on linux platform
-	if runtime.GOOS == "linux" && c.cpusetManager != nil {
-		// use the client configuration for reservable_cores if set
-		cores := c.config.ReservableCores
-		if len(cores) == 0 {
-			// otherwise lookup the effective cores from the parent cgroup
-			cores, _ = cgutil.GetCPUsFromCgroup(c.config.CgroupParent)
-		}
-		if cpuErr := c.cpusetManager.Init(cores); cpuErr != nil {
-			// If the client cannot initialize the cgroup then reserved cores will not be reported and the cpuset manager
-			// will be disabled. this is common when running in dev mode under a non-root user for example.
-			c.logger.Warn("failed to initialize cpuset cgroup subsystem, cpuset management disabled", "error", cpuErr)
-			c.cpusetManager = new(cgutil.NoopCpusetManager)
-		}
-	}
+	// startup the CPUSet manager
+	c.cpusetManager.Init()
+
```

Comment:
@shoenig I had to remove [this call](https://github.com/hashicorp/nomad/blob/main/client/client.go#L694-L695). It seems to be related to Nomad service disco checks?

Reply:
Yup, that checkstore.Init() is just checks related! (v1.4 only)